llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -5242,6 +5242,7 @@ bool MemProfContextDisambiguation::applyImport(Module &M) {
           assert(AI != FS->allocs().end());
           auto &AllocNode = *(AI++);
 
+#ifndef NDEBUG
           // Sanity check that the MIB stack ids match between the summary and
           // instruction metadata.
           auto MIBIter = AllocNode.MIBs.begin();
```

Comment:
Can we refactor this out to a static helper? I think it should be straightforward since the only live ins for this region are AllocNode and MemProfMD as far as I can tell.

Reply:
Done. There were a couple of other live in variables. I also updated one of the methods to take a const to enable using const for the new helper.